File: 7-kyu/string-packet-based-communications/python/solution.py
def communication_module(packet):
    header = packet[0:4]
    instruction = packet[4:8]
    data_p1 = int(packet[8:12])
    data_p2 = int(packet[12:16])
    footer = packet[16:]
    ret = ''
    if instruction == '0F12':
        ret = str(min(data_p1 + data_p2, 9999))
    elif instruction == 'B7A2':
        ret = str(max(data_p1 - data_p2, 0))
    elif instruction == 'C3D9':
        ret = str(min(data_p1 * data_p2, 9999))
    ret = header + 'FFFF' + ret.zfill(4) + '0000' + footer
    return ret

File: 7-kyu/string-packet-based-communications/c++/solution.cpp
std::string communicationModule(std::string packet)
{
    auto header = packet.substr(0, 4);
    auto instruction = packet.substr(4, 4);
    auto data1 = std::stoi(packet.substr(8, 4));
    auto data2 = std::stoi(packet.substr(12, 4));
    auto footer = packet.substr(16, 4);
    std::string retData = "";
    if (instruction == "0F12") retData = std::to_string(std::min(data1 + data2, 9999));
    else if (instruction == "B7A2") retData = std::to_string(std::max(data1 - data2, 0));
    else if (instruction == "C3D9") retData = std::to_string(std::min(data1 * data2, 9999));
    while (retData.length() < 4) retData = "0" + retData;
    return std::string(header + "FFFF" + retData + "0000" + footer);
}